Sprint 17

UdachiTomo opened the pull request.

UdachiTomo committed:
Работа по 17 спринту

UdachiTomo committed:
2. Начала реализации темной темы

UdachiTomo committed:
2.1 продолжение реализации темной темы

UdachiTomo committed:
2.2 реализована темная тема

UdachiTomo committed:
3.0 подключение cocoa pods и AppMetrica

UdachiTomo committed:
4.0 реализован экран статистики

UdachiTomo committed:
3.1 Реализована метрика

UdachiTomo committed:
4.1 устранены некоторые баги с отображением завершенных трекеров

UdachiTomo committed:
5. добавлены скриншот-тесты для светлой и темной темы

UdachiTomo committed:
3.1 Подгружен podifile

andrew-kuzn-yndx commented on AnalyticsService.swift:
```diff
@@ -0,0 +1,16 @@
+
+struct AnalyticsService {
+    static func activate() {
+        guard let configuration = YMMYandexMetricaConfiguration(apiKey: "6afd36d2-5535-4758-9b72-7e2cd23df2b7") else { return }
```
Различные apiKey стоит добавлять в отдельный файл типа Constants. Ты ведь делал такое в предыдущих спринтах, нужно поддерживать эту практику)

andrew-kuzn-yndx commented on CoreData/TrackerRecordStore.swift:
```diff
@@ -20,28 +67,87 @@ class TrackerRecordStore {
+        case .move:
+            guard let oldIndexPath = indexPath, let newIndexPath = newIndexPath else {
+                assertionFailure("move indexPath - nil")
+                return
```
Молодец, красиво сделал

andrew-kuzn-yndx commented on TrackerApp/CreateNewTrackerViewController/CreateNewTrackerViewController.swift:
```diff
@@ -104,6 +126,7 @@ final class CreateNewTrackerViewController: UIViewController, TrackerScheduleVie
         tableView.delegate = self
         tableView.layer.cornerRadius = 8
+        tableView.separatorColor = .ypGray
         tableView.separatorInset = UIEdgeInsets(top: 0, left: 345, bottom: 0, right: 345)
```
Для таких штук тоже хорошо бы делать Constants, но внутренний - где ты можешь в одном месте хранить значения констрейнтов. Это хорошая практика и сделает код более выигрышным

andrew-kuzn-yndx commented on TrackerApp/CreateNewTrackerViewController/CreateNewTrackerViewController.swift:
```diff
@@ -153,8 +176,25 @@ final class CreateNewTrackerViewController: UIViewController, TrackerScheduleVie
+                                            newEmoji: selectedEmoji,
+                                            newColor: selectedEmoji.color.hexString,
+                                            newSchedule: schedule,
+                                            categoryTitle: category?.title ?? "Без названия",
```
А тут, с точки зрения user expirience было бы здорово выдавать юзеру алерт, например. Мол "Вы уверены, что хотите создать трекер без категории"
Ну это так, как идея

andrew-kuzn-yndx commented on TrackerApp/TrackerCategory/MVVM/View/TrackerCategoryViewController.swift:
```diff
@@ -0,0 +1,202 @@
+    }
+    
+    private func actionSheet(categoryToDelete: TrackerCategory) {
+        let alert = UIAlertController(title: "Эта категория точно не нужна?",
```
Будет здорово, если потом найдешь время поместить все строки в Localizable

andrew-kuzn-yndx commented on TrackerApp/TrackerViewController/TrackerFilterViewController.swift:
```diff
@@ -0,0 +1,99 @@
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: TrackerCategoryTableViewCell.identifier, for: indexPath) as? TrackerCategoryTableViewCell else { return UITableViewCell() }
+        let filter = filters[indexPath.row]
+        cell.accessoryType = .none
+        cell.label.text = filter.rawValue
```
Так лучше не делать - управление аутлетами из другого класса плохая практика. Лучше сделать внутри ячейки функцию, типа configCell и уже в ней менять что нужно

UdachiTomo committed:
Исправлены замечания ревьвера, реализовано:
1. при нажатии на ячейку лейблы становятся видны
2. при редактировании трекера экран трекера заполнен данными

UdachiTomo committed:
3. корректно отображается экран статистики, если ни один трекер был не выполнен

UdachiTomo committed:
4. реализован градиент и рамка в StatisticViewController

UdachiTomo committed:
4.1 Отображается лейбл с днями при редактировании трекера

UdachiTomo committed:
исправлены замечания ревьювера:
1. убран показ лейбла с днями при создании трекера
2. убрана рамка обрамляющая количество выполненных дней на экране статистики, если выполненных трекеров еще не было
